tfhe/src/integer/gpu/mod.rs

```diff
@@ -8718,79 +9353,56 @@ pub(crate) unsafe fn cuda_backend_aggregate_one_hot_vector<
-        .as_ref()
+    let mut index_noise_levels = index_ct
         .info
         .blocks
         .iter()
         .map(|b| b.noise_level.0)
         .collect();
+    let mut ffi_index =
+        prepare_cuda_radix_ffi(index_ct, &mut index_degrees, &mut index_noise_levels);
 
-    let mut ffi_out_struct = prepare_cuda_radix_ffi(
-        lwe_array_out.as_ref(),
-        &mut ffi_out_degrees,
-        &mut ffi_out_noise_levels,
+    let mut match_degrees = vec![match_ct.0.ciphertext.info.blocks[0].degree.get()];
+    let mut match_noise_levels = vec![match_ct.0.ciphertext.info.blocks[0].noise_level.0];
+    let mut ffi_match = prepare_cuda_radix_ffi(
+        &match_ct.0.ciphertext,
+        &mut match_degrees,
+        &mut match_noise_levels,
     );
 
-    let mut ffi_in_degrees: Vec<Vec<u64>> = Vec::with_capacity(lwe_array_in_list.len());
-    let mut ffi_in_noise_levels: Vec<Vec<u64>> = Vec::with_capacity(lwe_array_in_list.len());
-
-    let ffi_in_structs: Vec<CudaRadixCiphertextFFI> = lwe_array_in_list
+    let mut ffi_selectors_degrees: Vec<Vec<u64>> = Vec::with_capacity(selectors.len());
+    let mut ffi_selectors_noise_levels: Vec<Vec<u64>> = Vec::with_capacity(selectors.len());
+    let ffi_selectors: Vec<CudaRadixCiphertextFFI> = selectors
         .iter()
         .map(|ct| {
-            assert_eq!(
-                streams.gpu_indexes[0],
-                ct.as_ref().d_blocks.0.d_vec.gpu_index(0)
-            );
-            let degrees: Vec<u64> = ct
-                .as_ref()
-                .info
-                .blocks
-                .iter()
-                .map(|b| b.degree.get())
-                .collect();
-            let noise_levels: Vec<u64> = ct
-                .as_ref()
-                .info
-                .blocks
-                .iter()
-                .map(|b| b.noise_level.0)
-                .collect();
-
-            ffi_in_degrees.push(degrees);
-            ffi_in_noise_levels.push(noise_levels);
+            let degrees = vec![ct.0.ciphertext.info.blocks[0].degree.get()];
+            let noise_levels = vec![ct.0.ciphertext.info.blocks[0].noise_level.0];
+            ffi_selectors_degrees.push(degrees);
+            ffi_selectors_noise_levels.push(noise_levels);
 
             prepare_cuda_radix_ffi(
-                ct.as_ref(),
-                ffi_in_degrees.last_mut().unwrap(),
-                ffi_in_noise_levels.last_mut().unwrap(),
+                &ct.0.ciphertext,
+                ffi_selectors_degrees.last_mut().unwrap(),
+                ffi_selectors_noise_levels.last_mut().unwrap(),
             )
         })
         .collect();
 
-    let num_input_ciphertexts = lwe_array_in_list.len() as u32;
-    let num_blocks = lwe_array_in_list[0]
-        .as_ref()
-        .d_blocks
-        .lwe_ciphertext_count()
-        .0 as u32;
-
     let mut mem_ptr: *mut i8 = std::ptr::null_mut();
-    scratch_cuda_aggregate_one_hot_vector_64(
```

Comment:
This scratch, the host & cleanup function should be removed from integer.h and from vector_find.cu as well please.